### PR TITLE
feat: remove tags filter from HowtoList

### LINKED ISSUE #1969: Release Categories Publicly
Time to release categories to the world 🎉
This feature been running in beta until now, we've now prepared how-tos + academy for its release. We're now ready to release categories publicly (out of beta-tester).

categories

 This requires hiding tags dropdown from how-to overview page.

Screenshot_2022-10-03_at_05_33_59

### PR DESCRIPTION
**Pull request by Sc4ramouche**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## TODOs

- [ ] update cypress tests
- [ ] figure out why filtering by category doesn't work in local env

## Description

Removes tags filter from the How-to's page. Changes:

### HowtoList

* removed `AuthWrapper` around categories filter so that it's available to everyone;
* `tags` in the SearchQuery are now ignored when opening the page;
* removed conditions for `selectedTags.length === 0` to show the "empty state";
* slightly changed imports order for the component.

### HowtoStore

* removed public `selectedTags` property;
* removed public `updateSelectedTags` property;
* now `filteredHowtos` are only filtered based on categories;

## Git Issues

Closes #1969 

## Screenshots/Videos

Screenshot 2022-10-03 at 16 32 30

### CODE REVIEW

**Comment by Sc4ramouche** [comment]
I'm not sure if it's an issue, but in my local env filtering by categories doesn't really yield any results:

filter-by-category

File: src/pages/Howto/Content/HowtoList/HowtoList.tsx [hunk 1]
```diff
@@ -105,25 +95,20 @@ export class HowtoList extends React.Component<any, IState> {
...
       this.injected.aggregationsStore.aggregations
 
-    const howtoItems = filteredHowtos.map((howto) => ({
+    const howtoItems = filteredHowtos.map((howto: IHowto) => ({
       ...howto,
       taglist:
         howto.tags &&
         Object.keys(howto.tags)
-          .map((t) => allTagsByKey[t])
+          .map((key) => allTagsByKey[key])
```
Comment (Sc4ramouche) [inline src/pages/Howto/Content/HowtoList/HowtoList.tsx:111]: concern: thought that key or k would make a bit more sense here, although let me know if you don't think so.

**Comment by Sc4ramouche** [comment]
Heja maintainers, it looks like Howto's in local env are coming from firebase. But it also looks like there we don't have "categories" attached to them. What would you suggest to test it properly in local env?
@thisislawatts @thisismattia

**Comment by thisislawatts** [comment]
Heya @Sc4ramouche, if you want to manipulate the data locally then I recommend using the running the Firebase Emulators locally within Docker. We have some documentation here outlining the steps:
https://onearmy.github.io/community-platform/Backend%20Development/firebase-emulators-docker

Thanks for picking up this item ✨

**Comment by Sc4ramouche** [comment]
> Heya @Sc4ramouche, if you want to manipulate the data locally then I recommend using the running the Firebase Emulators locally within Docker. We have some documentation here outlining the steps: https://onearmy.github.io/community-platform/Backend%20Development/firebase-emulators-docker
> 
> Thanks for picking up this item ✨

Cool, thanks for the hint 🙌 
I think now I have all the pieces to complete this one, I'll mark as "Ready for review" as soon as the data and cypress tests are in order.

**Comment by Sc4ramouche** [comment]
> Heya @Sc4ramouche, if you want to manipulate the data locally then I recommend using the running the Firebase Emulators locally within Docker. We have some documentation here outlining the steps: https://onearmy.github.io/community-platform/Backend%20Development/firebase-emulators-docker
> 
> Thanks for picking up this item ✨

Hey @thisislawatts, I have another question, sorry for bombarding with those:

I'm running the emulator according the docs linked above, but I don't get anything served on `localhost:4001`. It mentions that it may take a while to download the image. I waited for around 30 minutes yesterday – I still get `[emulators] Waiting for functions to be built...` message in my terminal. 

So, usually when you run those – does it come to an end? Just wanted to check what should I expect when I run `yarn start:emulated:docker`.

Screenshot 2022-10-05 at 13 13 15

**Comment by thisislawatts** [comment]
Thanks for the report @Sc4ramouche, I will take a look and see what could be going on here in the emulators. This looks like a bug.

**Comment by Sc4ramouche** [comment]
Thanks for the hint, once again! I'm no longer stuck with `"Waiting for functions to be built..."`.

Although now I have something else 😓
Screenshot 2022-10-14 at 23 47 40

Not sure what could've caused this, it feels like I miss something simple. Maybe I need to be logged in with Firebase somehow? I tried to upgrade firebase-functions package as well, didn't do the trick for me 🤔

There's also this chunk of warnings I get when I run `yarn start:emulated:docker`, attaching in case it could be relevant:

Screenshot 2022-10-14 at 23 51 05

Sorry for the mess here :( 

Although, since updating Firebase turned out to be tedious on my side, maybe we could collaborate on this and someone else could push a commit with updated seed data? I'd be happy to carry on and update the tests afterwards. It just feels that this could be a way to proceed and complete this one ✅ 

 @thisislawatts please let me know if you think this could work 🙏

**Comment by thisislawatts** [comment]
Thanks for the detailed report here @Sc4ramouche, it definitely looks like there is some additional work required here to make the emulators easier to work with. Flagging this for the attention of @ONEARMY/maintainers. In the meantime I am happy to help out.

**Review by thisislawatts — APPROVED** [review]

**Comment by thisislawatts** [comment]
@all-contributors add @Sc4ramouche for code

**Pull request merged**